routers/repo/issue.go

```diff
@@ -1618,7 +1618,7 @@ func UpdateIssueRef(ctx *context.Context) {
 		return
 	}
 
-	if !ctx.IsSigned || (!issue.IsPoster(ctx.User.ID) && !ctx.Repo.CanWriteIssuesOrPulls(issue.IsPull)) || issue.IsPull {
+	if !ctx.IsSigned || issue.IsPull || (!issue.IsPoster(ctx.User.ID) && !ctx.Repo.CanWrite(models.UnitTypeIssues)) {
```

Comment:
Having seen enough mistakes with permissions - and this is clearly the meat of the issue - we need to be more explicit with what we're doing.

```suggestion
	// Assert permissions:
	// 1. Only Signed in Users can change referred Branch.
	// 2. Only Issues - not Pulls have ref Branches.
	// 3. Omly the Issue Poster can change the Branch OR...
	// 4. Issue Writers may change the branch.
	//
	// Otherwise FORBIDDEN
	if !ctx.IsSigned || issue.IsPull || (!issue.IsPoster(ctx.User.ID) && !ctx.Repo.CanWrite(models.UnitTypeIssues)) {
```